Ansible playbook to install IPI on baremetal nodes

rlopez133 opened the pull request:
Ansible playbook that simplifies the deployment of IPI on baremetal. It is based on our manual installation guide found at: https://github.com/openshift-kni/baremetal-deploy/blob/master/install-steps.md . Please check out the README.md for instructions on how to use the playbook.

rlopez133 commented:
/assign @davidvossel

sreichar commented on ansible-ipi-install/roles/installer/tasks/02_get_oc.yml:
```diff
@@ -0,0 +1,27 @@
+---
+- name: Get the ocp client tar gunzip file
+  get_url:
+    url: "https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/{{ version }}/openshift-client-linux-{{ version }}.tar.gz"
```
for a verson, i'd like to specifi the latest-4.3 (or latest-4.4)  which works for the subdirectory but not the tar.
I think get the release name from the release.txt file on the Name: line

(could make this an issue /enhancement)

rlopez133 replied:
Can you add more details here on how to do that? Yes, I agree this could be an enhancement.

sreichar replied:
[sreichar@sreichar-t460p ~] $ curl -s https://mirror.openshift.com/pub/openshift-v4/clients/ocp-dev-preview/latest-4.3/release.txt | awk '/Name:/ {print $NF}'
4.3.0-0.nightly-2020-01-21-205041
[sreichar@sreichar-t460p ~] $

sreichar commented on ansible-ipi-install/roles/installer/tasks/02_get_oc.yml:
```diff
@@ -0,0 +1,27 @@
+  copy:
+    src: "{{ extract_dir }}/oc"
+    dest: /usr/local/bin/oc
+    owner: "{{ myuser }}"
```
shouldnt the owner/group be root?

sreichar commented on ansible-ipi-install/roles/installer/tasks/08_cleanup_sub_man_registeration.yml:
```diff
@@ -0,0 +1,8 @@
+    org_id: "{{ org_id }}"
+    state: absent
+  ignore_errors: yes
+
```
will every install be using subscription manager?   Should there be a when clause?

karmab replied:
this should subscribe the box when the credentials variables have been defined

sreichar commented on ansible-ipi-install/roles/installer/templates/metal3-config.j2:
```diff
@@ -0,0 +1,17 @@
+  name: metal3-config
+  namespace: openshift-machine-api
+data:
+  cache_url: ''
```
Why isnt' this specified?   Why not do cahcing?   Is this a later enhancement?

rlopez133 replied:
Adding as a feature enhancement: https://github.com/openshift-kni/baremetal-deploy/issues/131

sreichar commented on ansible-ipi-install/roles/installer/templates/metal3-config.j2:
```diff
@@ -0,0 +1,17 @@
+  ironic_endpoint: http://{{ prov_ip }}:6385/v1/
+  ironic_inspector_endpoint: http://{{ prov_ip }}:5050/v1/
+  provisioning_interface: {{ prov_nic }}
+  provisioning_ip: {{ prov_ip }}/24
```
I know the mask is 24 bit for now, but should we make this a variable since they are planning on making this flexible?

rlopez133 replied:
I think we can revisit this once it is an actual issue since provisioning_ip right now is not modifiable (yet).

sreichar commented on ansible-ipi-install/roles/node-prep/meta/main.yml:
```diff
@@ -0,0 +1,53 @@
+galaxy_info:
```
shoudln't this be filled out?

sreichar commented on ansible-ipi-install/roles/node-prep/tasks/02_bridge.yml:
```diff
@@ -0,0 +1,75 @@
+---
```
should we only do this if the bridge doesn't exist?

sreichar commented on ansible-ipi-install/roles/node-prep/tasks/09_create-install-config.yml:
```diff
@@ -0,0 +1,41 @@
+    cat /home/{{ myuser }}/.ssh/id_rsa.pub
+  register: sshkeycontents
+  when: sshkeypath.stat.exists == true
+
```
Shoudl you generate a key if it doesn't exist?

sreichar commented on ansible-ipi-install/roles/node-prep/tasks/main.yml:
```diff
@@ -0,0 +1,11 @@
+---
```
while I'd like to see tags a lot more places, would especially be nice here to controll execution of the top level tasks.

rlopez133 replied:
Feature enhancement #132

sreichar commented on ansible-ipi-install/roles/node-prep/tasks/10_power_off_cluster_servers.yml:
```diff
@@ -0,0 +1,16 @@
+---
```
I  know I got previous feedback - would it be better to install all packages at once instead of breaking out into multiple places

rlopez133 replied:
Honestly, I only did that for the purposely of clearly seeing why those packages were being installed versus someone asking later why are we installing X package?

sreichar commented on ansible-ipi-install/roles/node-prep/tasks/01_sub_man_register.yml:
```diff
@@ -0,0 +1,27 @@
+---
```
Not everybody is running a satellite or has permission to set a key at hosted,   do we know Verizon's story?

Think we need to make this task more optional.

rlopez133 replied:
It will be. For now its simple to either comment out the task or I can write a simple flag that lets you set it. I'm not trying to write for every use case as that be impossible :)

rlopez133 replied:
@sreichar - right now if activation_key or org_id is not defined, the sub manager task is skipped. Is this enough? Would you like to see an additional flag? true/false?

sreichar commented on ansible-ipi-install/roles/installer/tasks/main.yml:
```diff
@@ -0,0 +1,11 @@
+---
```
While I'd like to see lots more tags, think adding them to these top level task would help in controlling the flow when needed

rlopez133 replied:
agreed.

rlopez133 replied:
#132 Feature Enhancement

sreichar commented on ansible-ipi-install/roles/installer/tasks/08_cleanup_sub_man_registeration.yml:
```diff
@@ -0,0 +1,8 @@
+---
```
should we only unregistre if registered?

rlopez133 replied:
You wouldn't be able to get this far in the playbook if you were not registered (unless you happen to get all those dependencies some other way). Also since it is using a module, shouldn't matter.

sreichar commented on ansible-ipi-install/roles/installer/tasks/03_extract_installer.yml:
```diff
@@ -0,0 +1,16 @@
+    group: "{{ myuser }}"
+    mode: '0644'
+
+- name: Extracting the installer
```
Is there a reason not to put the installer in /usr/local/bin?

rlopez133 replied:
I'm just running it how we had it in our install guide. In that guide, we only put the oc binary in /usr/local/bin

karmab commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+# Provisioning IP address (default value)
+prov_ip: 172.22.0.3
+
+### Vars regarding install-config.yaml
```
i think it's debatible whether we want to be rendering install-config which tends to be super specific for each user

rlopez133 replied:
I think we can add additional logic for other options if required. However, I think having that file auto generated is a huge plus as creating it manually is a headache.

davidvossel replied:
yeah, as long as we have an escape hatch for providing your own install_config, I bet a lot of people will find autogenerating the install-config useful

rlopez133 replied:
I could see that being a feature enhancement. I'd like not to necessary delay getting this playbook inside this repo.

karmab commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+nummasters: ""
+apivip: ""
+ingressvip: ""
+ns1vip: ""
```
i think ns1vip and dnsvip are the same objects

rlopez133 replied:
They are. I need to remove one (ns1vip)

karmab commented on ansible-ipi-install/inventory/hosts:
```diff
@@ -0,0 +1,11 @@
+[masters]
```
people seem to prefer yaml inventory instead of ini format those days

rlopez133 replied:
Since we are talking preference here, does it matter?

karmab commented on ansible-ipi-install/playbook.yml:
```diff
@@ -0,0 +1,22 @@
+    pullsecret: 'ENTER SECRET HERE'
+  roles:
+    - role: node-prep
+      vars:
```
we'd want the specific variables to be defined outside of the playbook

rlopez133 replied:
All variables are defined and some default is set in the group_vars/all. However, for common variables that are likely to be modified or require to be modified it seemed the playbook.yml was the logical choice. I made my decision based on Ansible's doc: https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable

karmab commented on ansible-ipi-install/roles/installer/.travis.yml:
```diff
@@ -0,0 +1,29 @@
+---
+language: python
+python: "2.7"
```
shouldn't we use python3 ?

rlopez133 replied:
Interesting. I didn't create this file. It was created for me when I initiated the creation of the ansible role using ansible-galaxy cmd.

karmab commented on ansible-ipi-install/roles/installer/defaults/main.yml:
```diff
@@ -0,0 +1,2 @@
+---
+# defaults file for installer
```
no newline at end of file

karmab commented on ansible-ipi-install/roles/installer/tasks/02_get_oc.yml:
```diff
@@ -0,0 +1,27 @@
+    group: "{{ myuser }}"
+    mode: '0755'
+
+- name: "Untar the openshift-client-linux-{{ version }}.tar.gz"
```
as brillantly suggested by @e-minguez , we want to redownload oc client later on to match the release we target, to avoid potential issues if "latest" oc is not so latest (i think it's 4.2 since a couple of days)

rlopez133 replied:
I just realized that the code here wouldn't work for "latest" but only if we specified a specific version. This is because the naming scheme for latest is something like openshift-client-linux-4.4.0-0.nightly-2020-01-24-141203.tar.gz and the version was set to "latest" you'd get nothing back. Will fix logic here.

karmab commented on ansible-ipi-install/roles/installer/tasks/07_deploy_ocp.yml:
```diff
@@ -0,0 +1,5 @@
+---
+- name: IPI-INSTALL Deploy OpenShift Cluster
```
Remove IPI-INSTALL so that all task names follow a similar pattern (better for readability imho)

rlopez133 commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+extract_dir: /home/{{ myuser }}/extract
+
+# The user used to do the openshift-installation (requires passwordless sudo priv)
+myuser: ""
```
No we are not defaulting to the kni user. They just require a user with passwordless sudo priv and can set that user in their playbook.yml file (as shown in the example playbook.yml)

dcritch replied:
it can't be run as root? if it can, /home/root won't exist. should use use extract_dir: ~/extract ?

rlopez133 replied:
Ideally, I believe there should be a user with sudo priv available. I don't think we want to encourage usage of root user for installs.

karmab commented on ansible-ipi-install/roles/node-prep/templates/install-config.j2:
```diff
@@ -0,0 +1,41 @@
+  baremetal:
+    apiVIP: {{ apivip }}
+    ingressVIP: {{ ingressvip }}
+    dnsVIP: {{ ns1vip }}
```
should be dnsvip

dcritch replied:
does this even need to be specified? might be better to just resolve ns1, *.apps, etc. if something can be discovered, doing it that way can lead to less errors. you could also do an assert earlier on to ensure all the entries were created before hand, and bail otherwise

rlopez133 replied:
@dcritch to my knowledge it does. However, I'll defer to others whom may know more than I. I'll chat with you Monday to discuss further.

rlopez133 commented on ansible-ipi-install/inventory/hosts:
```diff
@@ -0,0 +1,11 @@
+worker-1 name=worker-1 role=worker user=admin password=password ipmi_address=192.168.1.5 mac=ec:f4:bb:da:32:28
+
+[provisioner]
+provisioner.example.com
```
Well you definitely need the FQDN for the provision host as your playbook is connecting here. Regarding masters and workers it doesn't matter because the information gathered in the inventory/hosts file is used to fill out the appropriate information in the install-config.j2 template file

rlopez133 replied:
You are correct that nothing in the playbooks ever directly executes on the master/worker nodes. However, we use this info to generate the install-config.yaml file. Since we are still talking about an inventory of hosts, does it not make more sense to keep it there versus as variables? If we decide to make them vars, do you for see then making a list for each item? i.e. ipmi_address, user, password, name etc?

davidvossel replied:
the only reason I considered them as vars is it centralizes everything related to a cluster in a Vars file, while the provisioner might be able to execute multiple cluster setups.

>  If we decide to make them vars, do you for see then making a list for each item? i.e. ipmi_address, user, password, name etc?

It's something worth experimenting with to see what formatting feels most natural. I'm not 100% certain what the constraints are in expressing something like this as vars.

rlopez133 replied:
@davidvossel I think its better to have them in the inventory file to be honest. Also speaking to @dcritch , I'm going to make some changes how I handle vars. Instead of having a group_vars/all file, I'm going to place all that info side a sample inventory file. This way people can use different inventory files to install environments without the need to change the playbook.yml

rlopez133 commented on ansible-ipi-install/playbook.yml:
```diff
@@ -0,0 +1,22 @@
+---
+- name: IPI on Baremetal Installation Playbook
+  hosts: provisioner
+  vars:
```
Every variable that has the potential to be modified is specified in the group_vars/all file. This allows anyone to see what is customizable. However, once someone wants to change a variable they can simply override the default in group_vars/all value in their playbook.yml. Just following best practice from: https://docs.ansible.com/ansible/latest/user_guide/playbooks_variables.html#variable-precedence-where-should-i-put-a-variable

rlopez133 commented on ansible-ipi-install/roles/installer/meta/main.yml:
```diff
@@ -0,0 +1,53 @@
+galaxy_info:
```
Yeah I can add default information in there. It doesn't effect the playbook though as I'm not using any meta.

karmab commented on ansible-ipi-install/roles/node-prep/templates/install-config.j2:
```diff
@@ -0,0 +1,41 @@
+      - name: {{ hostvars[host]['name'] }}
+        role: {{ hostvars[host]['role'] }}
+        bmc:
+          address: ipmi://{{ hostvars[host]['ipmi_address'] }}
```
it's possible the address to reach would use something else than ipmi?

rlopez133 replied:
I don't know --  @sreichar @schmaustech @dritch would you know if something other than ipmi would be used here?

rlopez133 replied:
Keeping this as a Feature Enhancement https://github.com/openshift-kni/baremetal-deploy/issues/133

karmab commented on ansible-ipi-install/roles/node-prep/templates/install-config.j2:
```diff
@@ -0,0 +1,41 @@
+          username: {{ hostvars[host]['user'] }}
+          password: {{ hostvars[host]['password'] }}
+        bootMACAddress: {{ hostvars[host]['mac'] }}
+        hardwareProfile: unknown
```
although i ignore whether it would work, a profile different than unknown might be needed for some specific workers(depending on their hardware)

rlopez133 replied:
@karmab - yes we probably need to discuss. I went by @e-minguez mentioning in a previous thread needing that definition. Aside from all that, I had discussions today with Julia Kreger and she mentioned that from her point of view the installer should only handle installations of the masters and workers should be added later. I plan to discuss with installer team their intentions regarding this.

karmab commented on ansible-ipi-install/roles/node-prep/vars/main.yml:
```diff
@@ -0,0 +1,6 @@
+---
+# vars file for node-prep
+package_list: ['firewalld','libvirt', 'qemu-kvm', 'mkisofs', 'python3-devel', 'jq', 'ipmitool']
```
i believe mkisofs is no longer required

rlopez133 replied:
> general comment: have you considered including an ansible.cfg file rather than have the user create it?

Yes! I plan on adding one.

karmab commented on ansible-ipi-install/roles/node-prep/vars/main.yml:
```diff
@@ -0,0 +1,6 @@
+package_list: ['firewalld','libvirt', 'qemu-kvm', 'mkisofs', 'python3-devel', 'jq', 'ipmitool']
+virtpool_packages: ['python3-libvirt','python3-lxml']
+nmcli_packages: ['NetworkManager-libnm','nm-connection-editor','libsemanage-python3','policycoreutils-python3']
+ipmi_power_packages: ['https://trunk.rdoproject.org/rhel8-master/deps/latest/Packages/python3-crypto-2.6.1-18.el8ost.x86_64.rpm','https://trunk.rdoproject.org/rhel8-master/deps/latest/Packages/python3-pyghmi-1.0.22-2.el8ost.noarch.rpm']
```
can't we gather those packages from an existing channel?

rlopez133 replied:
@karmab @sreichar @e-minguez  - From what I gathered, the ipmi_power_packages I want to install reside in the OpenStack 15 repository. I didn't know of another way to handle this so I'm really open to suggestions. I don't think we want to use pip? I also don't think we want to force people to enable an OpenStack subscription to get the packages.

rlopez133 commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+cmd: openshift-baremetal-install
+
+# The version of the openshift-installer 
+version: "4.3.0-0.nightly-2019-12-09-035405"
```
Your not fixed to this version, they can use any version they wish. I decided to put one that I know works as a default.

davidvossel replied:
maybe we shouldn't have a default so no one gets into the habit of not explicitly defining the version they want.

rlopez133 replied:
@davidvossel - thinking out loud, thoughts on defaulting to latest version? Problem is of what 4.3 or 4.4? I think the hard part right now is finding the version to use may not be a trivial task for the end user so supplying options may be worth it?

davidvossel replied:
i thought about the default being latest as well. My next thought was that someone would have a successful deployment with their config, and then try it again later not realizing that the version mutated underneath them. 

an explicit option to default version to latest might make sense. but like you pointed out... we'd have to pick out the latest build from a release series and I'm not sure how to auto detect that.

rlopez133 replied:
100% in agreement. I'm going to make this where explicit value needs to be provided for version and no default given.

davidvossel commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+org_id: ""
+
+# The directory used to store the cluster configuration files (install-config.yaml, pull-secret.txt, metal3-config.yaml)
+dir: /home/{{ myuser }}/clusterconfigs
```
to ensure this is unique (since one provisioning host might be used for multiple clusters) would if we used the clustername in this?

something like 

```
dir: /home{{ myuser }}/clusterconfigs/{{ cluster }}
```

rlopez133 replied:
Initially I was in agreement with this, but I'm going to put multiple cluster installs a feature enhancement for this playbook versus it delaying getting the playbook into this repo.

davidvossel commented on ansible-ipi-install/roles/installer/tasks/08_cleanup_sub_man_registeration.yml:
```diff
@@ -0,0 +1,8 @@
+---
+- name: IPI-INSTALL Unregister host via subscription-manager
```
why are we toggling subscription manger on/off during the installer steps, wouldn't this be in the node prep step?

davidvossel commented on ansible-ipi-install/roles/node-prep/tasks/02_bridge.yml:
```diff
@@ -0,0 +1,75 @@
+    stp: off
+    state: present
+
+- name: Create Bridge slave on {{ pub_nic }}
```
does this configuration logic around the public_nic interrupt the ansible ssh connection? i remember having to make special provisions to ensure some configs were done atomically across multiple files/cmds over ssh because it would cause my ssh connection to drop for awhile.

rlopez133 replied:
It actually doesn't interrupt the ansible ssh connection. Technically it is due to a bug in the nmcli.py file. Even though the nmcli module creates the bridges and slaves, when a file (ifcfg-*) already exists, and is modified, the module doesn't properly modify the file. Since it doesn't properly modify the file, I manually need to make edits to it. Since I make the edits to the file and the connection is active, I can then go ahead and use shell to ifdown and ifup the proper interfaces specific to the public nic. 

However, what I have is indeed a hack and in fact I included my own nmcli.py library file because there was bugs using the module against RHEL8 servers and it had no ip4_method available to it for manual connections. However, for our purposes and since I am supplying the module, it works as we'd want it to work (for now).

davidvossel replied:
> I can then go ahead and use shell to ifdown and ifup the proper interfaces specific to the public nic.

don't you at some point have to ifdown the public nic to pick up the changes though? I don't think I saw that happen.

rlopez133 replied:
its in there
```
- name: Get DHCP for baremetal bridge
  shell: |
    /usr/sbin/ifdown {{ pub_nic }}; /usr/bin/pkill dhclient ; /usr/sbin/ifdown baremetal ; /usr/sbin/ifup {{ pub_nic }}; /usr/sbin/ifup baremetal
```

davidvossel commented on ansible-ipi-install/roles/node-prep/tasks/09_create-install-config.yml:
```diff
@@ -0,0 +1,41 @@
+---
+- name: Check if Pull Secret is set
```
shouldn't install-config generation be a "per cluster" operation that takes place as part of the installer steps rather than during node-prep. It's feasible that one provisioning host can be used for multiple clusters.

rlopez133 replied:
Good question. I was always under the impression it was a 1 cluster per prov host but seems like that isn't the only use case.

davidvossel replied:
i think it's pretty likely we'll encounter a situation where we need one provisioning host to be capable of multiple cluster provisioning... It's definitely an outlier use case though.

rlopez133 replied:
I think multiple cluster is an interesting use case, but for the purposes of this playbook, I'm going to designate that a feature enhancement.

davidvossel commented on ansible-ipi-install/roles/node-prep/tasks/08_create_config_install_dirs.yml:
```diff
@@ -0,0 +1,12 @@
+---
+- name: IPI-INSTALL Create config dir and install dir
```
this seems more related to the installer steps instead of node-prep

rlopez133 replied:
I can see that, but I'm just creating directories. You can make the argument, I'm "prepping" so that the configs and install binaries can be set in place.

davidvossel commented on ansible-ipi-install/roles/node-prep/tasks/10_power_off_cluster_servers.yml:
```diff
@@ -0,0 +1,16 @@
+    state: present
+  with_items: "{{ ipmi_power_packages }}"
+
+- name: IPI-INSTALL Power OFF All Nodes
```
this this is also a "per-cluster" action, it probably belongs in the installer steps as well.

rlopez133 replied:
If we talking multiple clusters, yeah I would agree with that.

davidvossel commented on ansible-ipi-install/roles/node-prep/vars/main.yml:
```diff
@@ -0,0 +1,6 @@
+---
+# vars file for node-prep
+package_list: ['firewalld','libvirt', 'qemu-kvm', 'mkisofs', 'python3-devel', 'jq', 'ipmitool']
```
maybe we don't need mkisofs anymore? https://github.com/metal3-io/baremetal-operator/pull/310

rlopez133 replied:
yep will remove

rlopez133 commented on ansible-ipi-install/README.md:
```diff
@@ -0,0 +1,426 @@
+servers
+* Download a copy of your [Pull secret](https://cloud.redhat.com/openshift/install/metal/user-provisioned)
+* Append to the `pull-secret.txt` the [Pull secret](https://docs.google.com/document/d/1pWRtk7IbnfPo6cSDsopUMrxS22t3VJ2PuN39MJp9tHM/edit) 
+with access to `registry.svc.ci.openshift.org` and `registry.redhat.io`
```
@davidvossel good question. We probably need to remove this but I wasn't sure how we handle those other pullsecret registries? I guess just remove and keep that info internal or ?

davidvossel replied:
>  I guess just remove and keep that info internal or ?

i suppose so. i know people approaching this for the first time will get frustrated by that though. Maybe hint at needing to append credentials to the pull secret if you're accessing a private repo

rlopez133 commented on ansible-ipi-install/group_vars/all:
```diff
@@ -0,0 +1,47 @@
+cmd: openshift-baremetal-install
+
+# The version of the openshift-installer 
+version: "4.3.0-0.nightly-2019-12-09-035405"
```
I'm open. I like your suggestion of making it a required field. I also wouldn't mind seeing the default be latest version.

e-minguez commented on ansible-ipi-install/roles/installer/tasks/04_create_metal3.yml:
```diff
@@ -0,0 +1,26 @@
+  register: commit_id
+
+- name: Get RHCOS_URI
+  shell: |
```
Nit: I'd say to use ansible's filters and such to avoid doing so much shell commands (such as https://docs.ansible.com/ansible/latest/user_guide/playbooks_filters.html#json-query-filter )

rlopez133 replied:
Maybe you can assist me with this once we PR? I honestly haven't use ansible filters that much

e-minguez commented on ansible-ipi-install/roles/node-prep/tasks/01_sub_man_register.yml:
```diff
@@ -0,0 +1,27 @@
+  register: rhsmfile
+
+- name: "Check if any repos are enabled currently"
+  shell: 'subscription-manager repos --list-enabled | sed -ne "s/^Repo ID:[^a-zA-Z0-9]*\(.*\)/\1/p"'
```
I'm not sure if this is needed?

rlopez133 replied:
So the redhat_subscription module, if a playbook is run multiple times basically messes up the repos. In order to combat this, I do a simple check to see if any repos are enabled, if they are, then there is a likely possibility that you don't need to go through the registration process again. Is it perfect and catch every use case? Probably not. I'm open to cleaner ways but for our purposes this catches most use cases.

sreichar commented on ansible-ipi-install/inventory/hosts:
```diff
@@ -0,0 +1,11 @@
+[masters]
+master-0 name=master-0 role=master user=admin password=password ipmi_address=192.168.1.1 mac=ec:f4:bb:da:0c:58
```
should user and password be ipmi_user and ipmi_password
shoudl mac be provison_mac

rlopez133 commented:
/retest

slintes commented:
/retest

slintes commented:
`unit-tests` are working again, sorry for inconvenience.
we also try to limit running the test only when sth in the `features` path changes

rlopez133 commented:
> `unit-tests` are working again, sorry for inconvenience.
> we also try to limit running the test only when sth in the `features` path changes

No worries thank you for the help :)

rlopez133 commented:
Hey all - @sreichar @davidvossel @e-minguez @dcritch @karmab 
I've incorporated all your changes and here is a running list. Some items have been left has feature enhancements but should not block the merge of the code. Thank you to all who provided feedback, ansible snippets, etc :)

If you could please have a final review of the code so we can get this merged in. 

Thanks again! 

Roger

**Changes Incorporated**

- Fill out default values for meta/main.yml on each role
- version variable handles "latest-4.3, latest-4.4 and explicit value i.e. 4.3.0-0.nightly-2020-01-21-20504"
- All packages are installed in one location, 30_req_packages.yml in node-prep role
- 20_sub_man_register.yml only runs if activation_key and org_id are set in the inventory/hosts file otherwise skips using subscription manager
- Version variable no longer defaults to a value, but instead is an empty string. User must be explicit as to what version they want.
- Removed ns1vip variable as not used
- All modifiable variables are defined in inventory/hosts.sample file. No longer use a groups_var directory. Per consensus, it was recommended to have only one file that an end user would need to modify and that would be the inventory/hosts file
- Removed IPI-INSTALL on all task names
- Fix install-config.yml.j2 template replacing ns1vip with dnsvip as the variable. 
- Added sshKey into the install-config.yml.j2 template
- removed mkisofs package
- Added ansible.cfg file to repository
- Fixed README doc regarding "NIC interface names must be identical across all nodes"
- Updated README with latest changes, added Troubleshoot sections and Gotcha section
- Fixed the licensing in the meta/main.yml
- Removed myuser variable and now use ansible_user
- Made the UUID part of the pub_nic.j2 template
- Use slurp module instead of shell to capture sshKey
- Removed all unnecessary shell modules
- Fixed release image and release version using get uri module and regex (thanks @dcritch !)
- Handle whetehr you are using GA version or dev version using a variable labeled build 
- Using ansible_user_dir to locate user's home dir
- Change variables in the inventory/hosts.sample file regarding ipmi. Made them more descriptive

**Changes to be included as Enhanced Feature but not this release**
- Using tags across all tasks
- Change variable dir to handle for multi cluster environments
- ipmi power task to handle multiple clusters

karmab commented:
/lgtm

karmab commented:
/lgtm

karmab commented:
/approve
